### PR TITLE
[flutter_tools] pass through enable impeller flag to macOS.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Allow passing through the --enable-impeller flag to macOS.

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/desktop_device_test.dart [hunk 1]
```diff
@@ -300,6 +300,82 @@ void main() {
       ),
     );
   });
+
+  testWithoutContext('Device devices that support impeller pass through the enable-impeller flag', () async {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/desktop_device_test.dart:304]: Did you mean "Desktop devices"?

```suggestion
  testWithoutContext('Desktop devices that support impeller pass through the enable-impeller flag', () async {
```

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/desktop_device_test.dart:304]: lol yes, thanks

File: packages/flutter_tools/test/general.shard/desktop_device_test.dart [hunk 2]
```diff
@@ -300,6 +300,82 @@ void main() {
...
+          'FLUTTER_ENGINE_SWITCH_4': 'verify-entry-points=true',
+          'FLUTTER_ENGINE_SWITCHES': '4'
+        }
+      ),
+    ]);
+    final FakeDesktopDevice device = setUpDesktopDevice(
+      processManager: processManager,
+      supportsImpeller: true,
+    );
+    unawaited(Future<void>(() {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/desktop_device_test.dart:325]: Why is this unawaited?

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/desktop_device_test.dart:325]: Copy paste, removed

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
Technically this doesn't work until https://github.com/flutter/engine/pull/42639 rolls in, but its not harmful to land before that as it doesn't change behavior.

**Pull request merged**